packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -339,11 +340,16 @@ export class DocumentDeltaConnection
 	public submitSignal(content: string, targetClientId?: string): void {
 		this.checkNotDisposed();
 
-		if (targetClientId && this.details.supportedFeatures?.submit_signals_v2 !== true) {
-			throw new UsageError("Sending signals to specific client ids is not supported.");
+		// Check for server-side support of v2 signals
+		if (this.details.supportedFeatures?.submit_signals_v2 === true) {
+			const signal: ISentSignalMessage = { content };
+			if (targetClientId !== undefined) {
+				signal.targetClientId = targetClientId;
+			}
+			this.emitMessages("submitSignal", [signal]);
+		} else {
+			this.emitMessages("submitSignal", [[content]]);
```

Comment:
How do we provide the targeted signal experience for clients without the service having support?
Certainly, it seems possible that if all clients receive the signal, then it is possible that clients that were not meant to see the signal could avoid dealing with it.

Reply:
We could include the targetClientID in the signal content and then drop the signals on the recieving side if the targetClientID doesn't match. It seems there would be the problem with clients manually inserting a targetClientID property in the signal payload though which isn't desirable.

Reply:
If it happened at a higher level than here - perhaps where envelope is created as part of envelope - could it be backwards compatible?

Reply:
I made some changes here:
- Add targetClientId to SignalEnvelope when specified by sending client.
- When a signal is received by a client it checks if SignalMessage has a targetClientId (server supports targeted signals). 
- If targetClientId on SignalMessage is undefined, we then check if SignalEnvelope (content of signal message) contains a targetClientId (server does not support targeted signals but a targetClientId was specified by the sending client). 
- If targetClientId does not match the receiving client Id then we drop the signal os clients avoid dealing with signals that were not meant for them